### PR TITLE
Remove Indy/Vdrtools

### LINKED ISSUE #1250: Removal of legacy indy-sdk crates and dependencies
The indy-sdk has been deprecated in favor of multiple shared components (Aries Askar, anoncreds-rs, indy-vdr), which are already supported within Aries VCX via feature flags. The Aries VCX maintainers would like to remove these crates and the usage of indy-sdk in any of its forms in Aries VCX. This includes the removal of:
- `/misc/legacy/indyvdrtools` (essentially a modified fork of indy-sdk) -- replaced by aries-askar for storage requirements
- AnonCreds support via credx -- replaced by anoncreds-rs
- vdrtoolswallet (indy-sdk wallet) to aries-askar wallet migrator -- no longer necessary

This **does not** include the removal of the traits/interfaces in front of these tools--this will enable easier development of future implementations for different use cases while having common interfaces (different storage options or WASM compatible implementations for instance).

According to our present knowledge, these proposed crates/feature flags/tools to be removed were only being consumed by Absa, and Absa is no longer involved in the project. 

If any of these tools are important to your project or use case please let us know below, or reach out to one of the Aries VCX maintainers directly.

### PR DESCRIPTION
**Pull request by gmulhearn**
Closes #1250 

# Other changes
I believe switching the pipelines has caused a whole lot of new tests to run in askar mode, so there's plenty of new test failures.
* in aries_vcx, add support for expanding verkeys read from ledgers (abbreviated with ~ prefix, expanded into full verkey)
* rm mysql wallet tests for now, i don't believe these ever worked for askar, we may need to look into it if we want to support it
* askar wallet `verify` function only worked if the signer key is owned by the wallet; not useful for verify external signers
* change the from_seed to from_secret_bytes in askar key generation (similar to #1224 ), however i've taken the easy path out by just swapping it in-place
* small issues with the `unpack` alg, where the reported sender_vk was missing 1 step of base58 decoding.
* just using a dummy in-memory askar wallet for tests/agents currently. i left a TODO comment, but i think we can defer this for solving later, it will require consideration of how we want to make storage path configurable (previously we just let indy select some default location)
* pin the mediator build image to our globally used rust toolchain version (1.79)

### CODE REVIEW

**Comment by gmulhearn** [comment]
TODO - test against AATH

**Comment by gmulhearn** [comment]
todo-  check-aries-vcx-anoncreds is failing

**Comment by gmulhearn** [comment]
todo - test-unit failing

**Comment by gmulhearn** [comment]
partially uses https://github.com/hyperledger/aries-vcx/pull/1224

**Comment by gmulhearn** [comment]
AATH status:
- [ ] VCX <-> VCX

**Comment by gmulhearn** [comment]
FWIW: Anonyome was able to consume this update with no issues

**Review by JamesKEbert — APPROVED** [review]
Really awesome PR to see, thank you for the work that went into it! 

> rm mysql wallet tests for now, i don't believe these ever worked for askar, we may need to look into it if we want to support it

Seems appropriate to me--I'd leave it out until need arises or if someone volunteers to add it. If we think it's important we can create a GH issue to track it.

File: aries/aries_vcx_wallet/src/wallet/askar/askar_did_wallet.rs [hunk 1]
```diff
@@ -38,26 +37,34 @@ impl DidWallet for AskarWallet {
...
+            let pk = verkey.key();
+            if pk.len() != 32 {
+                return Err(VcxWalletError::InvalidInput(format!(
+                    "Invalid key length: {}",
+                    pk.len()
+                )));
+            }
+            bs58::encode(&pk[0..16]).into_string()
+        };
+
```
[the remark this answers is not in the document]
Comment (gmulhearn) [reply aries/aries_vcx_wallet/src/wallet/askar/askar_did_wallet.rs:57]: ye possibly. your fix takes it a little further with the ability to pass in a key. but maybe that could be a feature for another time. the core of the issue should be fixed

**Pull request merged**